python/pyarrow/tests/parquet/test_parquet_encryption.py

```diff
@@ -0,0 +1,248 @@
+            path, table.schema,
+            encryption_properties=file_encryption_properties) as writer:
+        writer.write_table(table)
+
+
+def read_encrypted_parquet(path, decryption_config,
+                           kms_connection_config, crypto_factory):
+    file_decryption_properties = crypto_factory.file_decryption_properties(
+        kms_connection_config, decryption_config)
+    assert(file_decryption_properties is not None)
+    meta = pq.read_metadata(
+        path, decryption_properties=file_decryption_properties)
+    assert(meta.num_columns == 3)
+    schema = pq.read_schema(
+        path, decryption_properties=file_decryption_properties)
+    assert(len(schema.names) == 3)
+
+    result = pq.ParquetFile(
+        path, decryption_properties=file_decryption_properties)
+    return result.read()
+
+
+@pytest.mark.parquet
+def test_encrypted_parquet_read_no_decryption_config(tempdir):
+    """Write an encrypted parquet, verify it's encrypted,
+    but then try to read it without decryption properties."""
+    with pytest.raises(IOError, match=r"no decryption"):
+        test_encrypted_parquet_write_read(tempdir)
+        path = tempdir / PARQUET_NAME
+        result = pq.ParquetFile(path)
+        assert(result is not None)
+
+
+@pytest.mark.parquet
+def test_encrypted_parquet_read_metadata_no_decryption_config(tempdir):
+    """Write an encrypted parquet, verify it's encrypted,
+    but then try to read its metadata without decryption properties."""
+    with pytest.raises(IOError, match=r"no decryption"):
+        test_encrypted_parquet_write_read(tempdir)
+        path = tempdir / PARQUET_NAME
+        meta = pq.read_metadata(path)
+        assert(meta is not None)
+
+
+@pytest.mark.parquet
+def test_encrypted_parquet_read_schema_no_decryption_config(tempdir):
+    """Write an encrypted parquet, verify it's encrypted,
+    but then try to read its schema without decryption properties."""
+    with pytest.raises(IOError, match=r"no decryption"):
+        test_encrypted_parquet_write_read(tempdir)
+        path = tempdir / PARQUET_NAME
+        schema = pq.read_schema(path)
+        assert(schema is not None)
+
+
+@pytest.mark.parquet
+def test_encrypted_parquet_write_no_col_key(tempdir):
+    """Write an encrypted parquet, but give only footer key,
+    without column key."""
+    with pytest.raises(RuntimeError, match=r"column_keys"):
+        path = tempdir / 'encrypted_table_no_col_key.in_mem.parquet'
+        table = DATA_TABLE
+
+        # Encrypt the footer with the footer key
+        encryption_config = pq.EncryptionConfiguration(
+            footer_key=FOOTER_KEY_NAME)
+
+        kms_connection_config = pq.KmsConnectionConfig(
+            custom_kms_conf={
+                FOOTER_KEY_NAME: FOOTER_KEY,
+                COL_KEY_NAME: COL_KEY,
+            }
+        )
+
+        def kms_factory(kms_connection_configuration):
+            return InMemoryKmsClient(kms_connection_configuration)
```

Comment:
It would be nice to also have a test with a raising KMS factory, and a KMS factory that doesn't return an actual KMS client.